google/generativeai/answer.py

```diff
@@ -107,11 +112,50 @@ def _make_grounding_passages(source: GroundingPassagesOptions) -> glm.GroundingP
     return glm.GroundingPassages(passages=passages)
 
 
+SourceNameType = [str, retriever_types.Corpus, glm.Corpus, retriever_types.Document, glm.Document]
+
+class SemanticRetrieverConfigDict(TypedDict):
+    source: SourceNameType
+    query: content_types.ContentsType
+    metadata_filter: Optional[Iterable[MetadataFilter]]
+    max_chunks_count: Optional[int] 
+    minimum_relevance_score: Optional[float]
+    
+SemanticRetrieverConfigOptions = Union[
+    SourceNameType,
+    SemanticRetrieverConfigDict,
+    glm.SemanticRetrieverConfig,
+]
+
+def _maybe_get_source_name(source)-> str|None:
+    if isinstance(source, str):
+        return source
+    elif isinstance(source, (retriever_types.Corpus, glm.Corpus, retriever_types.Document, glm.Document)):
+        return source.name
+    else:
+        return None
+
+
+
+def _make_semantic_retriever_config(
+    source: SemanticRetrieverConfigOptions,
+) -> glm.SemanticRetrieverConfig:
+    if isinstance(source, glm.SemanticRetrieverConfig):
+        return source
+
+    name = _maybe_get_source_name(source)
+    if name is not None:
+        source = {'source': name}
+
+    source['name'] = _maybe_get_source_name(source['source'])
+        
+    return glm.SemanticRetrieverConfig(source)
+
 def _make_generate_answer_request(
     *,
     model: model_types.AnyModelNameOptions = DEFAULT_ANSWER_MODEL,
     contents: content_types.ContentsType,
-    grounding_source: GroundingPassagesOptions,
+    grounding_source: GroundingPassagesOptions | SemanticRetrieverConfigOptions,
```

Comment:
```suggestion
    inline_passages: GroundingPassagesOptions | None,
    semantic_retreiver_config: GroundingPassagesOptions | None,
```

Reply:
And we just need a code block in here checking that exactly one of the two are set.